src/main/java/gavinchatbot/util/Parser.java

```diff
@@ -0,0 +1,80 @@
+package gavinchatbot.util;
+
+import java.time.LocalDate;
+
+import gavinchatbot.command.AddDeadlineCommand;
+import gavinchatbot.command.AddEventCommand;
+import gavinchatbot.command.AddToDoCommand;
+import gavinchatbot.command.Command;
+import gavinchatbot.command.DeleteCommand;
+import gavinchatbot.command.ExitCommand;
+import gavinchatbot.command.FindCommand;
+import gavinchatbot.command.ListCommand;
+import gavinchatbot.command.MarkCommand;
+import gavinchatbot.command.UnmarkCommand;
+
+/**
+ * Represents a parser that interprets user input and returns the appropriate command.
+ */
+public class Parser {
```

Comment:
I like how you have JavaDocs for all classes

Reply:
Thank you Sparsh